habitat_sim/robots/mobile_manipulator.py

```diff
@@ -0,0 +1,322 @@
+from typing import List, Optional, Tuple
+
+import attr
+import magnum as mn
+import numpy as np
+
+import habitat_sim
+from habitat_sim.robots.robot_interface import RobotInterface
+
+
+@attr.s(auto_attribs=True, slots=True)
+class MobileManipulatorParams:
+    """Data to configure a mobile manipulator.
+    :property arm_joints: The joint ids of the arm joints.
+    :property wheel_joints: The joint ids of the wheels. If the wheels are not controlled, then this should be None
+
+    :property arm_init_params: The starting joint angles of the arm. If None,
+        nothing is set.
+
+    :property ee_offset: The 3D offset from the end-effector link to the true
+        end-effector position.
+    :property ee_link: The Habitat Sim link ID of the end-effector.
+    :property ee_constraint: A (2,N) shaped array specifying the upper and
+        lower limits for each joint where N is the arm DOF.
+
+    :property arm_cam_offset_pos: The 3D offset of the arm camera from the
+        end-effector position.
+    :property head_cam_offset_pos: The 3D offset of the head camera from the
+        base of the robot.
+    :property head_cam_look_pos: The 3D offset of where the head should face,
+        relative to the head camera.
+
+    :property gripper_joints: The habitat sim joint ids of any grippers.
+    :property gripper_closed_state: All gripper joints need to achieve this
+        value for the gripper to be considered closed.
+    :property gripper_open_state: See gripper_closed_state.
+
+    :property arm_mtr_pos_gain: The position gain of the arm motor.
+    :property arm_mtr_vel_gain: The velocity gain of the arm motor.
+    :property arm_mtr_max_impulse: The maximum impulse of the arm motor.
+
+    :property wheel_mtr_pos_gain: The position gain of the wheeled motor (if
+        there are wheels).
+    :property wheel_mtr_vel_gain: The velocity gain of the wheel motor (if
+        there are wheels).
+    :property wheel_mtr_max_impulse: The maximum impulse of the wheel motor (if
+        there are wheels).
+    :property ctrl_freq: The number of control actions per second.
+    """
+
+    arm_joints: List[int]
+    wheel_joints: Optional[List[int]]
+
+    arm_init_params: Optional[List[float]]
+
+    ee_offset: mn.Vector3
+    ee_link: int
+    ee_constraint: np.ndarray
+
+    arm_cam_offset_pos: mn.Vector3
+    head_cam_offset_pos: mn.Vector3
+    head_cam_look_pos: mn.Vector3
+
+    gripper_joints: List[int]
+    gripper_closed_state: float
+    gripper_open_state: float
+
+    arm_mtr_pos_gain: float
+    arm_mtr_vel_gain: float
+    arm_mtr_max_impulse: float
+
+    wheel_mtr_pos_gain: float
+    wheel_mtr_vel_gain: float
+    wheel_mtr_max_impulse: float
+
+    ctrl_freq: int
+
+
+class MobileManipulator(RobotInterface):
+    """Robot with a controllable base and arm."""
+
+    def __init__(
+        self,
+        params: MobileManipulatorParams,
+        urdf_path: str,
+        sim: habitat_sim.simulator.Simulator,
```

Comment:
```suggestion
        sim: Simulator,
```

These two should make it so that you don't have to import simulator before robots.